File: C++/21-栈的压入、弹出序列.cpp
class Solution {
public:
    bool IsPopOrder(vector<int> pushV,vector<int> popV) {
        int len = pushV.size(), j = 0;
        vector<int> verify;
        for(int i = 0; i < pushV.size(); ++i){
            verify.push_back(pushV[i]);
            while(!verify.empty() && verify.back() == popV[j]){
                verify.pop_back();
                ++j;
            }
        }
        return verify.empty() ? true : false;
    }
};

File: Python/21-栈的压入、弹出序列.py
# -*- coding:utf-8 -*-
class Solution:
    def IsPopOrder(self, pushV, popV):
        # write code here
        verify = []
        j = 0
        for i in range(len(pushV)):
            verify.append(pushV[i])
            while verify and verify[-1] == popV[j]:
                verify.pop()
                j += 1
        return False if verify else True